cesm/flux_atmocn/flux_atmocn_driver_mod.F90

```diff
@@ -0,0 +1,131 @@
+
+    !--- input arguments --------------------------------
+    integer  , intent(in) :: logunit
+    integer  , intent(in) :: nMax        ! data vector length
+    integer  , intent(in) :: mask (nMax) ! ocn domain mask       0 <=> out of domain
+    logical  , intent(in) :: add_gusts
+    real(R8) , intent(in) :: zbot (nMax) ! atm level height (m)
+    real(R8) , intent(in) :: ubot (nMax) ! atm u wind (m/s)
+    real(R8) , intent(in) :: vbot (nMax) ! atm v wind (m/s)
+    real(R8) , intent(in) :: thbot(nMax) ! atm potential T (K)
+    real(R8) , intent(in) :: qbot (nMax) ! atm specific humidity (kg/kg)
+    real(R8) , intent(in) :: rainc(nMax) ! atm precip for convective gustiness (kg/m^3) - RBN 24Nov2008/MDF 31Jan2022
+    real(R8) , intent(in) :: rbot (nMax) ! atm air density (kg/m^3)
+    real(R8) , intent(in) :: tbot (nMax) ! atm T (K)
+    real(R8) , intent(in) :: pslv (nMax) ! atm sea level pressure(Pa)
+    real(R8) , intent(in) :: us   (nMax) ! ocn u-velocity (m/s)
+    real(R8) , intent(in) :: vs   (nMax) ! ocn v-velocity (m/s)
+    real(R8) , intent(in) :: ts   (nMax) ! ocn temperature (K)
+    real(R8) , intent(in) :: seq_flux_atmocn_minwind        ! minimum wind speed for atmocn (m/s)
+    integer  , intent(in) :: ocn_surface_flux_scheme
+
+    !--- output arguments -------------------------------
+    real(R8),intent(out)  ::  sen  (nMax)     ! heat flux: sensible (W/m^2)
+    real(R8),intent(out)  ::  lat  (nMax)     ! heat flux: latent (W/m^2)
+    real(R8),intent(out)  ::  lwup (nMax)     ! heat flux: lw upward (W/m^2)
+    real(R8),intent(out)  ::  evap (nMax)     ! water flux: evap ((kg/s)/m^2)
+    real(R8),intent(out)  ::  taux (nMax)     ! surface stress, zonal (N)
+    real(R8),intent(out)  ::  tauy (nMax)     ! surface stress, maridional (N)
+    real(R8),intent(out)  ::  tref (nMax)     ! diag:  2m ref height T (K)
+    real(R8),intent(out)  ::  qref (nMax)     ! diag:  2m ref humidity (kg/kg)
+    real(R8),intent(out)  :: duu10n(nMax)     ! diag: 10m wind speed squared (m/s)^2
+    real(R8),intent(out)  :: ugust_out(nMax)  ! diag: gustiness addition to U10 (m/s)
+    real(R8),intent(out)  :: u10res(nMax)     ! diag: gustiness addition to U10 (m/s)
+
+    real(R8),intent(out),optional :: ustar_sv(nMax) ! diag: ustar
+    real(R8),intent(out),optional :: re_sv   (nMax) ! diag: sqrt of exchange coefficient (water)
+    real(R8),intent(out),optional :: ssq_sv  (nMax) ! diag: sea surface humidity (kg/kg)
+    real(R8),intent(in) ,optional :: missval        ! masked value
+
+    !--- local variables --------------------------------
+    integer  :: n
+    real(R8) :: spval  ! local missing value
+    !--------------------------------------------------------------------------------
+
+    !!.................................................................
+    !! ocn_surface_flux_scheme = 0 : Large and Pond
+    !!                         = 1 : COARE algorithm
+    !!                         = 2 : UA algorithm
```

Comment:
(I know that referring to these by their numbers is a pre-existing thing that you've maintained, so this is more a nice-to-have thought rather than something I see as a requirement for this PR.)

It would be great if there could be named parameters for these values, like:

```Fortran
integer, parameter :: ocn_flux_scheme_undefined = -1
integer, parameter :: ocn_flux_scheme_large_and_pond = 0
integer, parameter :: ocn_flux_scheme_coare = 1
integer, parameter :: ocn_flux_scheme_ua = 2
```

So that, e.g., all the places that check:

```Fortran
if (ocn_surface_flux_scheme == 1) then
```

could be replaced with:

```Fortran
if (ocn_surface_flux_scheme == ocn_flux_scheme_coare) then
```